fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1338,6 +1339,107 @@ public void updateTableMeta(Database db, String tableName, Map<String, String> p
+                hasChanged = true;
+            }
+        }
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_BINLOG_TTL)) {
+            Long binlogTtl = Long.parseLong(properties.get(
+                    PropertyAnalyzer.PROPERTIES_BINLOG_TTL));
+            if (binlogTtl != newBinlogConfig.getBinlogTtlSecond()) {
+                newBinlogConfig.setBinlogTtlSecond(binlogTtl);
+                hasChanged = true;
+            }
+        }
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE)) {
+            Long binlogMaxSize = Long.parseLong(properties.get(
+                    PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE));
+            if (binlogMaxSize != newBinlogConfig.getBinlogMaxSize()) {
+                newBinlogConfig.setBinlogMaxSize(binlogMaxSize);
+                hasChanged = true;
+            }
+        }
+        if (!hasChanged) {
+            LOG.info("table " + olapTable.getName() +
+                    " binlog config is same as the previous version, so nothing need to do");
+            return true;
+        }
+
+        if (olapTable.getCurBinlogConfig() != null) {
+            LOG.info("update binlog config of table " + olapTable.getName() +
+                    ", the current binlog is :" + olapTable.getCurBinlogConfig().toString());
+        } else {
+            LOG.info("update binlog config of table " + olapTable.getName() +
+                    ", it has no binlog config previously");
+        }
+
+        db.writeLock();
+        // check for concurrent modifications by version
+        if (olapTable.getBinlogVersion() != newBinlogConfig.getVersion()) {
+            // binlog config has been modified,
+            // no need to judge whether the binlog config is the same as previous one,
+            // just modify even if they are the same
+            Map<String, String> newProperties = olapTable.getCurBinlogConfig().toProperties();
+            newProperties.putAll(properties);
+            newBinlogConfig.buildFromProperties(newProperties);
+        }
+        newBinlogConfig.incVersion();
+        try {
+            GlobalStateMgr.getCurrentState().modifyBinlogMeta(db, olapTable, newBinlogConfig);
+        } catch (Exception e) {
+            // defensive programming, it normally should not throw an exception,
+            // here is just to ensure that a correct result can be returned
+            isModifiedSuccess = false;
+        } finally {
+            db.writeUnlock();
+        }
+
+        // TODO optimize by asynchronous rpc
+        if (metaType != TTabletMetaType.DISABLE_BINLOG) {
+            try {
+                for (Partition partition : partitions) {
+                    updateBinlogPartitionTabletMeta(db, olapTable.getName(), partition.getName(), olapTable.getCurBinlogConfig(),
+                            TTabletMetaType.BINLOG_CONFIG);
+                }
+            } catch (DdlException e) {
+                LOG.warn(e.getMessage());
```

Comment:
This can be used to print the stack through log.warn(e), not just an error, which makes it easier to locate the problem.